guardian/editions
# article webview: implement scroll shadow in pure CSS/JS

## Summary

Get rid of passing the scroll position continuously from the webview into React Native, instead this makes it so the scroll marker is entirely implemented in the webview with no RN intervention. I'd expect this to be slightly more performant but that's too tiny to really measure. On the other hand, I think it's better because (1) we can have a subtle shadow rather than just a hairline, which is a common pattern for saying there is something above the fold and (2) we can have a smooth CSS transition between states.

## Test Plan

On iOS and on Android:

1. Open the app, open an article. Notice there is no border between the "slider" and the article content.
2. Scroll a bit. Notice the shadow appears, that indicate there is content above the fold. On Android that's super smooth, on iOS there's a delay. (Looks like [this changes for iOS 8+](https://dzone.com/articles/why-scroll-event-change-ios-8), but I think we're not using the right webview because of `useWebKit={false}`. We will have to fix this eventually anyway.)

<div>
<img width="300" alt="Screenshot 2019-11-01 at 15 23 35" src="https://user-images.githubusercontent.com/1733570/68036386-798fda80-fcbd-11e9-9808-f9863e3b9301.png">
<img width="300" alt="Screenshot 2019-11-01 at 15 23 40" src="https://user-images.githubusercontent.com/1733570/68036385-78f74400-fcbd-11e9-9553-99f78603504c.png">
</div>
